compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1044,8 +1044,17 @@ private BType inferTableMemberType(List<BType> memTypes, BLangTableConstructorEx
         BLangTableKeySpecifier keySpecifier = tableConstructorExpr.tableKeySpecifier;
         List<String> keySpecifierFieldNames = new ArrayList<>();
         Set<BField> allFieldSet = new LinkedHashSet<>();
+        List<BType> restFieldTypes = new ArrayList<>();
+
         for (BType memType : memTypes) {
-            allFieldSet.addAll(((BRecordType) memType).fields.values());
+            BRecordType member = (BRecordType) memType;
+            allFieldSet.addAll(member.fields.values());
+            if (!member.sealed) {
+                BType restFieldType = member.restFieldType;
+                if (isUniqueType(restFieldTypes, restFieldType)) {
```

Comment:
Just for my understanding, 
is there a difference if we use a `LinkedHashSet<BType>` instead of `isUniqueType()` check here?

Reply:
If we use a hashset, two instances of say `new BType(TypeTags.INT)` will not be considered as duplicates.  isUniqueType() checks through the `BSameTypeVisitor`

Reply:
I see. Thanks for the clarification. :)

Say we have types, `(int|string)` and `int`, where two `int`s are from two `new BType(TypeTags.INT)` instances.
Now we do the `isUniqueType()` check here and conclude they are unique. 

Then we create the union with `BUnionType.create(null, types);`. Inside the `BUnionType.create()` we seem to be using `LinkedHashSet<BType>` and no `isUniqueType()` check for the flatten members. So the resulant union will be `(int|string|int)`.

Just wondering whether we should do the following here?
``` 
            if (restFieldType.tag == TypeTags.UNION) {
                ((BUnionType) restFieldType).getMemberTypes().forEach(type -> {
                    if (isUniqueType(restFieldTypes, type)) {
                        types.add(type);
                    }
                });
            } else if (isUniqueType(restFieldTypes, restFieldType)) {
                types.add(restFieldType);
            }

```

Reply:
`new BType(TypeTags.INT)` was provided as an example to say the object references would be different; anyway, the int would come from the symbol table; where the references would be anyway the same. let's have a offline chat regarding the exact requirement